### PR TITLE
Docs/tutorial cleanup

### PR DESCRIPTION
**Pull request by filbabic**
Docs improvements, API cleanup, some smaller missing features...

https://user-images.githubusercontent.com/17215808/201074625-8916248e-cbad-4b93-b73b-ddd18bf850a0.mp4

### CODE REVIEW

File: docusaurus/docs/Android/04-guides/01-call-lifecycle.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+To create the client, simple call `StreamVideo.createCallClient()`, with the corresponding parameters, like so:
+
+```kotlin
+val callClient = streamVideo.createCallClient(
+    signalUrl,
+    userToken,
+    iceServers,
+    credentialsProvider
+)
```
Comment (skydoves) [inline docusaurus/docs/Android/04-guides/01-call-lifecycle.mdx]: What do you think about adding some comments for each parameter? 😄

File: docusaurus/docs/Android/04-guides/01-call-lifecycle.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+* Browsing a list of calls
+* Creating new calls or inviting people to existing
+* Ringing a group in chat-based applications or features
+* Generating call URLs for invites
+
+Additionally, the instance creates a Web Socket connection that listens to events like `CallCreated`, which you can use to show incoming call UI to your users or trigger custom business logic.
+
+If your app relies on such events at its core, we recommend to initialize it **as soon as your user is authenticated**. Once you do that, make sure to **keep it in a high level component**, like the `Application` class, so you can share the instance.
+
```
Comment (skydoves) [inline docusaurus/docs/Android/04-guides/01-call-lifecycle.mdx]: Do we need to describe how users can initialize `StreamVideo` with some example code, like the `Application` class?

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/calling/audio/AudioCalling.kt [hunk 3]
```diff
@@ -1,98 +0,0 @@
...
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package io.getstream.video.android.compose.ui.components.calling.audio
-
-import androidx.compose.foundation.Image
```
Comment (filbabic) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/calling/audio/AudioCalling.kt:19]: @skydoves You built these components when I was away but from what I've seen, they weren't used anywhere and some of them were not fully done.

We do need to expose these but I'm not sure if it's just a duplication, given that we have `OutgoingCallScreen` and `IncomingCallScreen` components too

File: app/src/main/kotlin/io/getstream/video/android/app/ui/components/UserItem.kt [hunk 4]
```diff
@@ -101,7 +101,7 @@ fun UserItem(
                         .size(40.dp)
                         .align(alignment = Alignment.Center),
                     imageUrl = credentials.image,
-                    initials = credentials.name
+                    initials = credentials.name.split(" ").joinToString { it }
```
Comment (TGazica) [inline app/src/main/kotlin/io/getstream/video/android/app/ui/components/UserItem.kt:104]: Shoudln't it be `.joinToString { it.first() }`?

File: app/src/main/kotlin/io/getstream/video/android/app/ui/call/CallActivity.kt [hunk 5]
```diff
@@ -107,7 +108,13 @@ class CallActivity : AppCompatActivity() {
...
         val isInitialized = callViewModel.isVideoInitialized.value
         if (isInitialized) return
-        callViewModel.connectToCall()
+        callViewModel.connectToCall(
+            CallSettings(
+                audioOn = false,
+                videoOn = true,
+                speakerOn = false
+            )
+        )
```
Comment (TGazica) [inline app/src/main/kotlin/io/getstream/video/android/app/ui/call/CallActivity.kt:117]: If the VM is handling business logic shouldn't this be handled through the VM? We can expose the default parameters for the call in the constructor and change them through setters.

File: docusaurus/docs/Android/02-client/02-stream-video.mdx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+val credentialsProvider = AuthCredentialsProvider(
+    apiKey = "<your-api-key>",
+    userToken = token, // the token from the BE
+    user = user // the User object with extra information
+)
+
+val streamVideo = StreamVideoBuilder(
+    context = this,
+    credentialsProvider = credentialsProvider,
+    loggingLevel = LoggingLevel.BODY, // we support
```
Comment (TGazica) [inline docusaurus/docs/Android/02-client/02-stream-video.mdx:17]: `// we support` should have been `// we support logging on different levels` maybe?

File: docusaurus/docs/Android/04-guides/02-call-lifecycle.mdx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+> **Note**: While the diagram shows the lifecycle of only one StreamVideo client and CallClient, you can create many instances during a user session, if the user logs in and out multiple times, and joins and leaves multiple calls.
+
+Let's see when it's best to initialize and dispose of these clients.
+
+### StreamVideo
+
+The `StreamVideo` client **requires an authenticated user** to be created and worked with.
+
+If your has steps where the user doesn't have to be authenticated, like anonymous browsing, exploring the app after installing, but before creating an account, or similar, then we **don't recommend** initializing the `StreamVideo` client right away.
```
Comment (TGazica) [inline docusaurus/docs/Android/04-guides/02-call-lifecycle.mdx]:
```suggestion
If your application has steps where the user doesn't have to be authenticated, like anonymous browsing, exploring the app after installing, but before creating an account, or similar, then we **don't recommend** initializing the `StreamVideo` client right away.
```

File: dogfooding/src/main/java/io/getstream/video/android/dogfooding/CallActivity.kt [hunk 8]
```diff
@@ -98,7 +103,13 @@ class CallActivity : AppCompatActivity() {
...
         val isInitialized = callViewModel.isVideoInitialized.value
         if (isInitialized) return
-        callViewModel.connectToCall()
+        callViewModel.connectToCall(
+            CallSettings(
+                audioOn = false,
+                videoOn = true,
+                speakerOn = false
+            )
+        )
```
Comment (TGazica) [inline dogfooding/src/main/java/io/getstream/video/android/dogfooding/CallActivity.kt:112]: Same as the app module comment.

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/CallContent.kt [hunk 9]
```diff
@@ -59,14 +59,14 @@ import androidx.compose.ui.unit.dp
...
-import io.getstream.video.android.compose.ui.components.MainStage
+import io.getstream.video.android.compose.ui.components.CallControls
+import io.getstream.video.android.compose.ui.components.participants.internal.MainStage
 import io.getstream.video.android.model.CallParticipantState
 import io.getstream.video.android.model.state.StreamCallState
 import io.getstream.video.android.viewmodel.CallViewModel
 
 @Composable
-internal fun VideoCallContent(
+public fun CallContent(
```
Comment (TGazica) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/CallContent.kt:69]: kdocs

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/CallControls.kt [hunk 10]
```diff
@@ -45,7 +45,7 @@ import androidx.compose.ui.unit.dp
 import io.getstream.video.android.model.Call
 
 @Composable
-public fun CallOptions(
+public fun CallControls(
```
Comment (TGazica) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/CallControls.kt:48]: kdocs

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/CallParticipant.kt [hunk 11]
```diff
@@ -27,12 +27,12 @@ import io.getstream.video.android.compose.theme.VideoTheme
 import io.getstream.video.android.compose.ui.components.avatar.UserAvatar
 import io.getstream.video.android.compose.ui.components.video.VideoRenderer
 import io.getstream.video.android.model.Call
-import io.getstream.video.android.model.CallParticipant
+import io.getstream.video.android.model.CallParticipantState
 
 @Composable
-public fun ParticipantItem(
+public fun CallParticipant(
```
Comment (TGazica) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/CallParticipant.kt:33]: kdocs

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/CallParticipants.kt [hunk 12]
```diff
@@ -30,7 +30,7 @@ import io.getstream.video.android.compose.theme.VideoTheme
 import io.getstream.video.android.model.Call
 
 @Composable
-public fun ParticipantsContent(
+public fun CallParticipants(
```
Comment (TGazica) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/CallParticipants.kt:33]: kdocs

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/LocalVideo.kt [hunk 13]
```diff
@@ -34,12 +34,12 @@ import androidx.compose.ui.unit.dp
 import io.getstream.video.android.compose.theme.VideoTheme
 import io.getstream.video.android.compose.ui.components.video.VideoRenderer
 import io.getstream.video.android.model.Call
-import io.getstream.video.android.model.CallParticipant
+import io.getstream.video.android.model.CallParticipantState
 
 @Composable
-public fun FloatingParticipantItem(
+public fun LocalVideo(
```
Comment (TGazica) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/LocalVideo.kt:40]: kdocs

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/internal/MainStage.kt [hunk 14]
```diff
@@ -14,23 +14,23 @@
...
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import io.getstream.video.android.compose.ui.components.participants.ParticipantsContent
+import io.getstream.video.android.compose.ui.components.participants.CallParticipants
 import io.getstream.video.android.model.Call
 import org.webrtc.SurfaceViewRenderer
 
 @Composable
-public fun MainStage(
+internal fun MainStage(
```
Comment (TGazica) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/internal/MainStage.kt:28]: are we going to document internal implementations like we do in the chat project?

File: app/src/main/kotlin/io/getstream/video/android/app/ui/components/UserItem.kt (hunk 4 above)
Comment (filbabic) [reply app/src/main/kotlin/io/getstream/video/android/app/ui/components/UserItem.kt:104]: Yeah fair enough - missed this bit!

File: docusaurus/docs/Android/02-client/02-stream-video.mdx (hunk 6 above)
Comment (filbabic) [reply docusaurus/docs/Android/02-client/02-stream-video.mdx:17]: yes

Comment (filbabic) [reply docusaurus/docs/Android/02-client/02-stream-video.mdx:17]: can fix in another docs pass

File: app/src/main/kotlin/io/getstream/video/android/app/ui/components/UserItem.kt (hunk 4 above)
Comment (filbabic) [reply app/src/main/kotlin/io/getstream/video/android/app/ui/components/UserItem.kt:104]: let us not forget this

**Pull request merged**